### PR TITLE
VG-67-Create-a-database-that-stores-ordered-products

### PR DESCRIPTION
**Pull request by Rishad11**
Related to issue

* Installed mongodb dependencies and packages.
* Updated docker-compose.yml to include a mongodb service within docker.
* Connected the backend with the database:
This was done with a retry mechanism to ensure that the express server waits for a successful connection to the database before it starts.

### CODE REVIEW

**Review by Danback — APPROVED** [review]
Ran compose and tested through postman, works as intended, great job!

File: docker-compose.yml
Comment (Demppas) [inline docker-compose.yml:1]: allt ser bra ut! Kanske uppdatera Version till en mer aktuell version? 3.8 eller 3.9 ? 
Bra att uppdatera då det är några versioner bakom.

**Review by marcusju23 — APPROVED** [review]
Seems to work as intended, great work! 👍

File: src/router/server.js
Comment (Demppas) [inline src/router/server.js:1]: Du använder även en äldre version av connect: 

> const client = await MongoClient.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
> console.log("Connected to MongoDB!");
> return client.db();
> 

Om du uppdaterar din version så kommer du behöva ändra denna genom att skapa en ny MongoClient och sedan kalla på den:

> const client = new MongoClient(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
> await client.connect();
> console.log("Connected to MongoDB!");
> return client.db();

File: docker-compose.yml
Comment (Rishad11) [reply docker-compose.yml:1]: Oh my bad, I took the latest version of the docker compose tool and not the compose file version. My mistake, I'll change it to the latest compose file version.

**Pull request merged**